internal/pkg/skuba/etcd/member_test.go

```diff
@@ -0,0 +1,176 @@
+)
+
+func Test_RemoveMember(t *testing.T) {
+	fakeCfgFiles := map[string][]byte{
+		"KubeletComponentConfig": []byte(`
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+`),
+		"ClusterConfiguration": []byte(`
+apiVersion: kubeadm.k8s.io/v1beta1
+kind: ClusterConfiguration
+kubernetesVersion: v1.13.0
+apiServer:
+  extraArgs:
+    advertiseAddress: 1.2.3.4
+`),
+	}
+
+	fakeConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kubeadm-config",
+			Namespace: metav1.NamespaceSystem,
+		},
+		Data: map[string]string{
+			kubeadmconstants.ClusterConfigurationConfigMapKey:     string(fakeCfgFiles["ClusterConfiguration"]),
+			kubeadmconstants.KubeletBaseConfigurationConfigMapKey: string(fakeCfgFiles["KubeletComponentConfig"]),
+		},
+	}
+
+	fakeDaemonSet := &appsv1.DaemonSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DaemonSet",
+			APIVersion: "app/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: metav1.NamespaceSystem,
+			Name:      "cilium",
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "fake",
+							Image: "fake",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	fakeJobSpec := batchv1.JobSpec{
+		Template: corev1.PodTemplateSpec{
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:    "fake",
+						Image:   "fake",
+						Command: []string{"/bin/bash"},
+					},
+				},
+				RestartPolicy: corev1.RestartPolicyNever,
+				Volumes:       []corev1.Volume{},
+				NodeSelector: map[string]string{
+					"kubernetes.io/hostname": "fake",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Operator: corev1.TolerationOpExists,
+					},
+				},
+			},
+		},
+	}
+
+	fakeWorker := corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "worker",
+		},
+	}
+
+	fakeMaster := corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "master",
+			Labels: map[string]string{"node-role.kubernetes.io/master": ""},
+		},
+	}
+
+	tests := []struct {
+		errExpected bool
+		errMessage  string
+		name        string
+	}{
+		{
+			name:        "should remove etcd member from etcd cluster",
+			errExpected: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			clientset := fake.NewSimpleClientset(&corev1.NodeList{Items: []corev1.Node{fakeMaster, fakeWorker}})
+
+			clientset.CoreV1().ConfigMaps(metav1.NamespaceSystem).Create(fakeConfigMap)
+			clientset.AppsV1().DaemonSets(metav1.NamespaceSystem).Create(fakeDaemonSet)
+
+			controlPlaneNodes, _ := kubernetes.GetControlPlaneNodes(clientset)
+			hashTarget := fmt.Sprintf("%x", sha1.Sum([]byte(fakeWorker.ObjectMeta.Name)))
+			hashExecutor := fmt.Sprintf("%x", sha1.Sum([]byte(controlPlaneNodes.Items[0].ObjectMeta.Name)))
+			job := fmt.Sprintf("caasp-remove-etcd-member-%.10s-from-%.10s", hashTarget, hashExecutor)
```

Comment:
We could make `removeMemberFromJobName` public instead of duplicating this logic here.

Reply:
Thanks, I'll enhance in another PR.

Reply:
Actually, scratch my comment, the function is on the same package (missed this) so we can call it as it is and we don't need to make it public.